src/Init/Core.lean

```diff
@@ -1922,7 +1929,7 @@ represents an element of `Squash α` the same as `α` itself
 `Squash.lift` will extract a value in any subsingleton `β` from a function on `α`,
 while `Nonempty.rec` can only do the same when `β` is a proposition.
 -/
-def Squash (α : Sort u) := Quot (fun (_ _ : α) => True)
+def Squash (α : Sort u) := Quotient (trueSetoid α)
```

Comment:
Please explain in the PR description why this is preferable (if possible *independently* of Mathlib's `Trunc`).